app/models/manageiq/providers/vmware/infra_manager/metrics_capture.rb

```diff
@@ -357,37 +335,33 @@ def perf_capture_intervals(mors, interval_name)
     interval_by_mor
   end
 
-  def perf_capture_counters(interval_by_mor)
-    _log.info("Capturing counters...")
-
-    counters_by_mor = {}
-
-    # Query Vim for all of the available metrics and their associated counter info
-    interval_by_mor.each do |mor, interval|
-      counters_by_mor[mor] = self.class.avail_metrics_for_entity(target.ext_management_system, @perf_vim_hist, mor, interval, @perf_intervals[interval.to_s])
-    end
-
-    _log.info("Capturing counters...Complete")
-    counters_by_mor
-  end
-
-  def perf_build_query_params(interval_by_mor, counters_by_mor, start_time, end_time)
+  def perf_build_query_params(interval_by_mor, counter_info, start_time, end_time)
     _log.info("Building query parameters...")
 
     params = []
     interval_by_mor.each do |mor, interval|
-      counters = counters_by_mor[mor]
-      next if counters.empty?
-
       st, et = Metric::Helper.sanitize_start_end_time(interval, @perf_intervals[interval.to_s], start_time, end_time)
 
+      perf_metric_id_set = []
+      counter_info.map do |counter_id, _counter_info|
+        aggregate_instances = "".freeze
+        all_instances       = "*".freeze
```

Comment:
I wonder if this should just be moved out to a constant (or even better move the array of the 2 out to a constant).